Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java

```diff
@@ -418,6 +418,12 @@ public void swapSlotWithProduction(final String subscriptionId, final String app
         slot.swap("production");
     }
 
+    public void deleteDeploymentSlotNode(final String subscriptionId, final String appId,
+                                         final String slotName) throws IOException {
+        final WebApp app = AuthMethodManager.getInstance().getAzureClient(subscriptionId).webApps().getById(appId);
+        app.deploymentSlots().deleteByName(slotName);
```

Comment:
check whether there will be potential NPE for `AuthMethodManager.getInstance().getAzureClient(subscriptionId).webApps().getById(appId)`
If so, suggest to perform NPE check over all these methods.

Reply:
This code snippet appears in many places, I will check.